commit 7a136c3bdab8cb8b75189c7370272261e9a14c0d
Author: Christian Glindkamp <christian.glindkamp@taskit.de>
Date:   Mon Jul 25 14:36:14 2011 +0200

    Add Sparsemem support for AT91SAM9G45

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 9adc278..a0138c0 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -296,6 +296,9 @@ config ARCH_AT91
 	select HAVE_CLK
 	select CLKDEV_LOOKUP
 	select ARM_PATCH_PHYS_VIRT if MMU
+	select ARCH_SPARSEMEM_ENABLE
+	select ARCH_SELECT_MEMORY_MODEL
+
 	help
 	  This enables support for systems based on the Atmel AT91RM9200,
 	  AT91SAM9 and AT91CAP9 processors.
diff --git a/arch/arm/mach-at91/include/mach/memory.h b/arch/arm/mach-at91/include/mach/memory.h
index 401c207..2d86fd4 100644
--- a/arch/arm/mach-at91/include/mach/memory.h
+++ b/arch/arm/mach-at91/include/mach/memory.h
@@ -23,4 +23,32 @@
 
 #include <mach/hardware.h>
 
+#ifndef __ASSEMBLY__
+
+#if defined(CONFIG_ARCH_AT91SAM9G45) || defined(CONFIG_ARCH_AT91SAM9M10)
+/*
+ * Non-linear mapping like so:
+ * phys       => virt
+ * 0x70000000 => 0xc0000000
+ * 0x20000000 => 0xd0000000
+ */
+
+#define __phys_to_virt(p)   \
+            (((p) & 0x0fffffff) + (((p) & 0x40000000) ? 0xc0000000 : 0xd0000000))
+
+#define __virt_to_phys(v)   \
+            (((v) & 0x0fffffff) + (((v) & 0x10000000) ? 0x20000000 : 0x70000000 ))
+
+#define NODE_MEM_SIZE_BITS     28
+#define MAX_PHYSMEM_BITS       32
+#define SECTION_SIZE_BITS      28 /* 256 Mb */
+#define HIGH_MEMORY_VIRT       0xe0000000
+
+extern unsigned long __pv_phys_offset;
+#define PHYS_OFFSET __pv_phys_offset
+
+#endif
+
+#endif
+
 #endif
diff --git a/arch/arm/mm/init.c b/arch/arm/mm/init.c
index c19571c..23c8195 100644
--- a/arch/arm/mm/init.c
+++ b/arch/arm/mm/init.c
@@ -390,7 +390,11 @@ void __init bootmem_init(void)
 	 */
 	arm_bootmem_free(min, max_low, max_high);
 
+#ifdef HIGH_MEMORY_VIRT
+	high_memory = HIGH_MEMORY_VIRT;
+#else
 	high_memory = __va(((phys_addr_t)max_low << PAGE_SHIFT) - 1) + 1;
+#endif
 
 	/*
 	 * This doesn't seem to be used by the Linux memory manager any
