diff --git a/arch/arm/mach-at91/Kconfig b/arch/arm/mach-at91/Kconfig
index d8fd5af..bed7e38 100644
--- a/arch/arm/mach-at91/Kconfig
+++ b/arch/arm/mach-at91/Kconfig
@@ -414,6 +414,16 @@ config MACH_STAMP9G45
 	  evaluation board.
 	  <http://www.taskit.de/en/>
 
+if MACH_STAMP9G45
+config BCH_CONST_PARAMS
+	def_bool y
+config BCH_CONST_M
+	int
+	default 13
+config BCH_CONST_T
+	int
+	default 4
+endif
 endif
 
 # ----------------------------------------------------------
